climbing_stairs/climbing_stairs.py

```diff
@@ -2,13 +2,25 @@
 
 import sys
 
+
 def climbing_stairs(n, cache=None):
-  pass 
+    hops = 0
+    ways = 0
+    if n <= 0:
+        return hops
+    if n > 3:
+        if n % 3 == 0:
+            hops += 3
+            print("hop")
+            climbing_stairs((n - hops))
+    return hops
```

Comment:
I would start this off with some simple cases, for example 
`if n < 0:
  return 0`
and one for if (elif) `n == 0`. Then you'll need to check if a cache exists and if there is an entry for `n`, then return that entry. Otherwise, if there is no cache, create it and the cache entry at `n` is equal to recursively calling the function itself with `n-1`, `n-2`, and `n-3` for the permutation possibilities. Finally, return the cache entry at `n`. Beej went over this in lecture yesterday, I believe.